Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,283 @@
+
+        headerContainer.snp.makeConstraints { make in
+            make.height.equalTo(58)
+        }
+
+        let titles = viewControllers.map {
+            $0.category
+        }
+        tabBar = ScrollableTextTabBarControl(sections: titles, padding: 16)
+        tabBar.directionalLayoutMargins = NSDirectionalEdgeInsets(top: 0, leading: 16, bottom: 0, trailing: 16)
+        tabBar.addTarget(self, action: #selector(manualScroll), for: .valueChanged)
+        tabBar.backgroundColor = .white
+
+        tabBar.snp.makeConstraints { make in
+            make.height.equalTo(40)
+        }
+
+        tabBarStack = UIStackView(arrangedSubviews: [headerContainer, tabBar])
+        tabBarStack.bringSubview(toFront: headerContainer)
+        tabBarStack.axis = .vertical
+        view.addSubview(tabBarStack)
+        view.bringSubview(toFront: tabBarStack)
+    }
+
+    private func createMenuStack() {
+        menuItemsView = ExpandedMenuItemView(frame: .zero, eatery: eatery, views: originalViews, allMenuItems: allItems)
+        menuItemsView.backgroundColor = .white
+        view.addSubview(menuItemsView)
+        view.sendSubview(toBack: menuItemsView)
+
+        tabBarStack.snp.makeConstraints { make in
+            make.top.equalToSuperview()
+            make.leading.trailing.equalToSuperview()
+        }
+
+        menuItemsView.snp.makeConstraints { make in
+            make.top.equalTo(tabBar.snp.bottom)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+    }
+
+    /*  This function manages the automatic scroll that occurs when a user presses a button in the tabbar
+        to go to a specific expanded menu category. This can happen in 2 ways:
+        1. The user presses the button after already scrolling past the top of this ViewController
+        2. The user presses the button before scrolling past this view controller
+        (i.e. scrollView is near the top of CampusMenuViewController)
+        The first operation is very simple, as visible by the "if" statement below. However, the second
+        (occurs in "else) is much more complicated because we have to split the automatic scroll operation
+        into two pieces: scrolling to the top of this view controller and scrolling to the target.  If we
+        don't do this, the tabBarStack will scroll unevenly with the rest of the scroll view.   */
+    @objc func manualScroll() {
+        let minOffset: CGFloat = 4
+        let currentPos = scrollView?.contentOffset.y ?? 0
+        let viewControllerPosOnParent = view.frame.origin.y
+        let viewPos = originalViews[tabBar.selectedSegmentIndex].frame.minY
+        let targetPos = viewPos - scrollOffset + viewControllerPosOnParent + minOffset
+        let topOfSelf = viewControllerPosOnParent - scrollOffset
+        let duration = animationDuration(from: currentPos, to: targetPos, withConst: 0.03)
+
+        isManualScrolling = true
```

Comment:
Is `isManualScrolling` still useful? Seems like it's set to true here then set to false again at the end of this function